apps/pre-award/lambdas/application-deadline-reminder/helpers/aws_extended_client.py

```diff
@@ -0,0 +1,173 @@
+        self,
+        aws_access_key_id=None,
+        aws_secret_access_key=None,
+        region_name="eu-west-2",
+        endpoint_url=None,
+        large_payload_support=None,
+        always_through_s3=None,
+        **kwargs,
+    ):
+        self.large_payload_support = large_payload_support
+        self.always_through_s3 = always_through_s3
+
+        if aws_access_key_id and aws_secret_access_key:
+            self.sqs_client = boto3.client(
+                "sqs",
+                aws_access_key_id=aws_access_key_id,
+                aws_secret_access_key=aws_secret_access_key,
+                region_name=region_name,
+                endpoint_url=endpoint_url,
+                **kwargs,
+            )
+            self.s3_client = boto3.client(
+                "s3",
+                aws_access_key_id=aws_access_key_id,
+                aws_secret_access_key=aws_secret_access_key,
+                region_name=region_name,
+                endpoint_url=endpoint_url,
+            )
+        else:
+            """
+            if 'aws_access_key_id' and 'aws_access_key_id' are not provided make sure to provide
+            'AWS_ACCESS_KEY_ID' and 'AWS_SECRET_ACCESS_KEY' with environment variables
+            """
+            self.sqs_client = boto3.client(
+                "sqs",
+                region_name=region_name,
+                endpoint_url=endpoint_url,
+                **kwargs,
+            )
+            self.s3_client = boto3.client(
+                "s3",
+                region_name=region_name,
+                endpoint_url=endpoint_url,
+            )
+
+    def submit_single_message(
+        self,
+        queue_url,
+        message,
+        extra_attributes: dict = None,
+        message_group_id=None,
+        message_deduplication_id=None,
+    ):
+        sqs_message_attributes = {
+            "message_created_at": {
+                "StringValue": str(datetime.now()),
+                "DataType": "String",
+            },
+        }
+        message_body, message_attributes = self._store_message_in_s3(
+            message, sqs_message_attributes, extra_attributes
+        )
+        # add extra message attributes (if provided)
+        if extra_attributes:
+            for key, value in extra_attributes.items():
+                message_attributes[key] = value
+
+        response = self.sqs_client.send_message(
+            QueueUrl=queue_url,
+            MessageBody=message_body,
+            MessageAttributes=message_attributes,
+            MessageGroupId=message_group_id,
+            MessageDeduplicationId=message_deduplication_id,
+        )
+        # Check if the delete operation succeeded, if not raise an error?
+        status_code = response["ResponseMetadata"]["HTTPStatusCode"]
+        if status_code != 200:
+            logging.error(
+                f"submit_single_message failed with status code {status_code}."
+            )
+        message_id = response["MessageId"]
```

Comment:
Will `response["MessageId"]` be available if `status_code != 200` ? If not, perhaps we should return `None` (otherwise it will throw a `KeyError` exception)